### ISSUE TITLE
[BUG] Hide played items from playlist broken

### ISSUE THREAD

**Issue by chuckkay** [yuliskov/SmartTube#3955]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/yuliskov/SmartTube/issues) or [closed](https://github.com/yuliskov/SmartTube/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://github.com/yuliskov/SmartTube#faq) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.

### Affected version

23.88

### Device Type

Smart TV/Box

### Affected Android

Firetv tv

### Steps to reproduce the bug

Add videos to watch later playlist, watch some videos, leave playlist and come back 

### Actual behavior

watched videos are still in the platlist even with a full red progress bar under the watched videos

### Additional information

Its Been happening for the past 2 weeks

**Comment by chuckkay** [comment]
Bump, am I alone with this issue?

**Comment by rguenther1209** [comment]
me as well - nvidia shield - been like this for ages, very frustrating

**Comment by AndrewGable** [comment]
I see this as well, the described bug is in https://github.com/yuliskov/SmartTube/issues/3758 too.

**Comment by AndrewGable** [comment]
Seems like it's this line, https://github.com/yuliskov/SmartTube/blob/master/common/src/main/java/com/liskovsoft/smartyoutubetv2/common/app/models/playback/controllers/VideoStateController.java#L622

Asked in Telegram for help on how to build as gradle sync isn't working on the latest Android Studio.

**Comment by retrocaster** [comment]
Your reproduction steps don't say anything about removing the video after watching it. Are you expected it to remove automatically after watching?

**Comment by AndrewGable** [comment]
Also wanted to note that removing a video from "watch later" playlist manually works, so ideally we could re-use that logic for the automatic removing.

**Comment by AndrewGable** [comment]
> Are you expected it to remove automatically after watching?

Yes I would expect this when the "Hide watched videos from Watch later playlist" is turned "ON". 

Watched videos are removed from the Watch later playlist once you navigate away from the watch later playlist and return.

**Comment by rguenther1209** [comment]
yea thats what the option says 'hide unwanted content' i founf for it to
work well i had to keep cache clean
